source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -700,16 +702,141 @@ def apply_final_journey_filters_post_finalize(response_list, request):
+    }
+
+    attractivity = {
+        "stop_point:IDFM:monomodalStopPlace:58498": 9,
+        "stop_point:IDFM:22041": 6,
+        "stop_point:IDFM:463281": 6,
+        "stop_point:IDFM:412988": 4,
+        "stop_point:IDFM:412987": 4,
+    }
+
+    olympics_durations = [j.duration for r in responses for j in r.journeys if 'olympics' in j.tags]
+    average_duration = sum(olympics_durations) / float(len(olympics_durations))
+
+    retained_journey = None
+
+    found_sps = []
+    for r in responses:
+        for j in r.journeys:
+            if 'olympics' not in j.tags:
+                continue
+            pt_extremity = get_journey_pt_extremity(j, request.get('clockwise'))
+            if pt_extremity.uri not in found_sps:
+                found_sps.append(pt_extremity.uri)
+            interval = intervals.get(pt_extremity.uri)
+            print(pt_extremity.uri, interval, average_duration)
+            if not interval.includes(average_duration):
+                j.tags.append("to_delete")
+            else:
+                if retained_journey is None:
+                    retained_journey = j
+                elif retained_journey.duration > j.duration:
+                    retained_journey.tags.append("to_delete")
+                    retained_journey = j
+                else:
+                    j.tags.append("to_delete")
+
+    if retained_journey is not None:
+        return
+
+    found_sps.sort(key=lambda sp: attractivity[sp])
+    best_attractivity = found_sps[-1]
+    for r in responses:
+        for j in r.journeys:
+            if 'olympics' not in j.tags:
+                continue
+            pt_extremity = get_journey_pt_extremity(j, request.get('clockwise'))
+            if pt_extremity.uri != best_attractivity:
+                continue
+            if retained_journey is None or retained_journey.duration > j.duration:
+                retained_journey = j
+
+    retained_journey.tags.remove("to_delete")
+
+
+def filter_olympics_journeys_v2(responses, request):
+    fake_fallback_durations = {
+        # RER E
+        "stop_point:IDFM:monomodalStopPlace:58498": 20,
+        # Metro 12
+        "stop_point:IDFM:22041": 600,
+        "stop_point:IDFM:463281": 600,
+        # T3b
+        "stop_point:IDFM:412988": 1000,
+        "stop_point:IDFM:412987": 1000,
+    }
+    best = None
+    for r in responses:
+        for j in r.journeys:
+            if 'olympics' not in j.tags:
+                continue
+            pt_extremity = get_journey_pt_extremity(j, request.get('clockwise'))
+            fake_fallback_duration = fake_fallback_durations.get(pt_extremity.uri)
+            fake_duration = j.duration - j.sections[-1].duration + fake_fallback_duration
+            if best is None:
+                best = (j, fake_duration)
+                continue
+
+            if best[1] > fake_duration:
+                best[0].tags.append("to_delete")
+                best = (j, fake_duration)
```

Comment:
be careful that the `best` journey you select at the end may have a 'to_delete' tag that was set in another filtering function

Reply:
I removed `to_delete` and I tag the best journey with `best_olympics`

Reply:
If we do this, we might need a bonus evolution of the playground that make the `best_olympic` game logo glow :sparkles: (to highlight it compare to others `olympic` journeys)